s3.go

```diff
@@ -156,6 +156,9 @@ func (s *S3Bucket) Query(q dsq.Query) (dsq.Results, error) {
 		return nil, fmt.Errorf("s3ds: filters or orders are not supported")
 	}
 
+	// S3 store a "/foo" key as "foo" so we need to trim the leading "/"
+	q.Prefix = strings.TrimPrefix(q.Prefix, "/")
```

Comment:
Is this not a problem in other functions? Get, put, delete? What about the keys returned by the query?

Reply:
It's not a problem with the other functions, `ListObjectsV2` just stand out in weirdness.

Reply:
I just checked again, the returned keys have the `/` prefix as it should.

Reply:
Got it. Well, in that case, LGTM. Thanks!